modules/openxr/openxr_api.cpp

```diff
@@ -2274,33 +2274,41 @@ String OpenXRAPI::action_set_get_name(RID p_action_set) {
 	return action_set->name;
 }
 
-bool OpenXRAPI::action_set_attach(RID p_action_set) {
-	ActionSet *action_set = action_set_owner.get_or_null(p_action_set);
-	ERR_FAIL_NULL_V(action_set, false);
+bool OpenXRAPI::attach_action_sets(const Vector<RID> &p_action_sets) {
+	ERR_FAIL_COND_V(session == XR_NULL_HANDLE, false);
 
-	if (action_set->is_attached) {
-		// already attached
-		return true;
-	}
+	XrActionSet *action_handles = new XrActionSet[p_action_sets.size()];
```

Comment:
I'm probably missing it but are you freeing `action_handles` anywhere?

Reply:
Alternatively you can use:
```
Vector<XRActionSet> action_handles;
action_handles.resize(p_action_sets.size())
```
which will be freed automatically.

But now I'm nitpicking :)

Reply:
Well using Godot templates that do memory management in a safe way is definitely not a nitpick, it's our policy :)
There's rarely any reason to use `new`.